### PR TITLE
Give user a link to create PR after push

### PR DESCRIPTION
**Pull request by JulienTant**
## New Feature

This PR offers a link to create new pull request when we push to a branch which is not the default branch.
If a PR already exists, it just give the link to the existing PR.

### Output of a push when no active PR exists:
```
[branch2/test be40717] test
 1 file changed, 1 deletion(-)
Counting objects: 3, done.
Writing objects: 100% (3/3), 255 bytes | 255.00 KiB/s, done.
Total 3 (delta 0), reused 0 (delta 0)
remote:
remote: Create a new pull request for 'branch2/test':
remote:   http://localhost:3000/test1/test/compare/master...branch2%2Ftest
remote:
To http://localhost:3000/test1/test.git
   0c5683f..be40717  branch2/test -> branch2/test
```

### Output of a push when an active PR exists:
```
[branch2/test 453d638] test
 1 file changed, 1 insertion(+)
Counting objects: 3, done.
Writing objects: 100% (3/3), 255 bytes | 255.00 KiB/s, done.
Total 3 (delta 0), reused 0 (delta 0)
remote:
remote: Visit the existing pull request:
remote:   http://localhost:3000/test1/test/pulls/6
remote:
To http://localhost:3000/test1/test.git
   be40717..453d638  branch2/test -> branch2/test
```

### CODE REVIEW

**Comment by JulienTant** [comment]
Now i wanna talk about some technical decision i took :

1) In case of errors, I log but does nothing because I don't want this feature to block anything.
2) I created 2 internal endpoints, but I though about an option where i could make only one that gave me everything I needed : the PR default branch + the PR if it exists. 
3) About the fork limitation, If you guys think it's better to create the PR in the "base" repository, then maybe it's better to consider to use what i described above to have the url generation out of the cmd/hook.go file

**Comment by lunny** [comment]
gitea_no_gcc should be removed.
I think the PR should be in base repository.

**Comment by lunny** [comment]
And you also should check if user has permission to visit and create pull request.

**Comment by lafriks** [comment]
I also agree that PR should be for base repository as that is default behaviour from UI

**Comment by JulienTant** [comment]
Ok I will first add the fork thing.

> And you also should check if user has permission to visit and create pull request.

I'm not sure how to do that. I took a look at the "web" router to access to PR creation page and I found this :
https://github.com/go-gitea/gitea/blob/7781e8cef2dfe73d71be7804f4c5a7c5f1995d31/routers/repo/issue.go#L79-L91

Is that what I must us implement ?

**Comment by JulienTant** [comment]
- Forks are now created for the base repository.
- Ensure that the "target" repository has enabled PR.

**Review by lunny — CHANGES_REQUESTED** [review]

File: modules/private/repository.go
Comment (JulienTant) [reply modules/private/repository.go:5]: I changed my PHPStorm template for the project, thanks.

**Comment by JulienTant** [comment]
Hi guys, any news on this ?

**Comment by lafriks** [comment]
@JulienTant it looks good, nice work but I want to test it first before I give my approval

**Comment by JulienTant** [comment]
Of course. I tried to lot of cases but if you find anything i'll be happy to fix/change things.

**Comment by JulienTant** [comment]
up?

**Comment by lafriks** [comment]
We are feature freeze currently so need to wait until v1.6.0-rc1 is released

**Review by lunny — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**
